src/orca-jedi/increment/Increment.cc

```diff
@@ -55,24 +64,495 @@ Increment::Increment(const Geometry & geom,
+      "orcamodel::Increment::read not implemented";
   throw eckit::NotImplemented(err_message, Here());
 }
-// -----------------------------------------------------------------------------
-Increment::Increment(const Increment & other)
-  : geom_(other.geom_), vars_(other.vars_), time_(other.time_),
-    incrementFields_()
-{
+
+void Increment::write(const eckit::Configuration & conf) const {
   std::string err_message =
-      "orcamodel::Increment::copy constructor not implemented";
+      "orcamodel::Increment::write not implemented";
   throw eckit::NotImplemented(err_message, Here());
 }
 
+void Increment::print(std::ostream & os) const {
+  double sumx2;
+  double sumx;
+  double min;
+  double max;
+  int valid_points;
+
+  oops::Log::trace() << "Increment(ORCA)::print starting" << std::endl;
+
+  os << "Increment valid at time: " << validTime() << std::endl;
+  os << std::string(4, ' ') << vars_ <<  std::endl;
+  os << std::string(4, ' ') << "atlas field:" << std::endl;
+  for (atlas::Field field : incrementFields_) {
+    std::string fieldName = field.name();
+    std::tie(valid_points, sumx2, sumx, min, max) = stats(fieldName);
+    os << std::string(8, ' ') << fieldName <<
+          " num: " << valid_points <<
+          " mean: " << std::setprecision(5) << sumx/valid_points <<
+          " rms: " << sqrt(sumx2/valid_points)  <<
+          " min: " << min << " max: " << max << std::endl;
+  }
+  oops::Log::trace() << "Increment(ORCA)::print done" << std::endl;
+}
+
+std::tuple<int, double, double, double, double>
+      Increment::stats(const std::string & fieldName) const {
+  int valid_points = 0;
+  double sumx = 0;
+  double sumx2 = 0;
+  double min = 1e30;
+  double max = -1e30;
+
+  auto field_view = atlas::array::make_view<double, 2>(
+      incrementFields_[fieldName]);
+  oops::Log::trace() << "Increment(ORCA):stats" << std::endl;
+  auto ghost = atlas::array::make_view<int32_t, 1>(
+      geom_->mesh().nodes().ghost());
+  atlas::field::MissingValue mv(incrementFields()[fieldName]);
+  bool has_mv = static_cast<bool>(mv);
+  for (atlas::idx_t j = 0; j < field_view.shape(0); ++j) {
+    for (atlas::idx_t k = 0; k < field_view.shape(1); ++k) {
+      if (!ghost(j)) {
+        if (!has_mv || (has_mv && !mv(field_view(j, k)))) {
+          if (field_view(j, k) > max) { max=field_view(j, k); }
+          if (field_view(j, k) < min) { min=field_view(j, k); }
+          sumx += field_view(j, k);
+          sumx2 += field_view(j, k)*field_view(j, k);
+          ++valid_points;
+        }
+      }
+    }
+  }
+  return std::make_tuple(valid_points, sumx2, sumx, min, max);
+}
+
+double Increment::norm() const {
+  int valid_points;
+  int valid_points_all = 0;
+  double sumx2all = 0;
+  double sumx2;
+  double sumx;
+  double min;
+  double max;
+
+  for (atlas::Field field : incrementFields_) {
+    std::string fieldName = field.name();
+    std::tie(valid_points, sumx2, sumx, min, max) = stats(fieldName);
```

Comment:
I am thinking that this is a large number of ordered parameters - its easy to make a mistake and create a hidden runtime bug. I am thinking it would be clearer if `stats` returned a struct with these attributes instead of a tuple? Alternatively, maybe if stats is only used here you could define it as a lambda function local to `Increment::norm`?